src/jp/hidetobara/pasteltouch/PastelTouchActivity.java

```diff
@@ -0,0 +1,535 @@
+					.copy(Config.ARGB_8888, true);			
+    	}
+    	catch(Exception ex)
+    	{
+			dp(ex.getMessage());
+			_Bitmap = null;
+    	}
+    }
+
+    private Matrix makeFitMatrix(Bitmap b, Display d)
+    {
+    	Matrix m = new Matrix();
+    	int bh = b.getHeight();
+    	int bw = b.getWidth();
+
+    	boolean isRotated = false;
+    	if(d.getWidth() < d.getHeight() && b.getWidth() > b.getHeight()) isRotated = true;
+    	if(d.getWidth() > d.getHeight() && b.getWidth() < b.getHeight()) isRotated = true;
+    	if(isRotated)
+    	{
+    		m.postRotate(90.0f);
+    		bh = b.getWidth();
+    		bw = b.getHeight();
+    	}
+		float sw = (float)d.getWidth() / (float)bw;
+		float sh = (float)d.getHeight() / (float)bh;
+		float scale = sw > sh ? sh : sw;
+		if( scale > 1.0f ) scale = 1.0f;
+		m.postScale(scale, scale);
+    	return m;
+    }
+    
+    private void assignBitmap()
+    {
+    	if(_Bitmap == null || _Canvas == null) return;
+
+    	_Canvas.setBitmap(_Bitmap);	//Bitmapの適応
+		_Canvas.setActivity(this);	//activityの登録
+		_Canvas.reinitialize();		//初期化
+		_Canvas.setBitmapFit(); 	//表示をFitさせる
+    }
+    
+    public void registerFileToGallery(File file)
+    {
+    	if( file == null ) return;
+    	
+        String filepath = file.getPath();
+        String filename = file.getName();
+        String suffix   = Utility.getSuffix(file);
+        String mimetype = "";
+        
+        if ((suffix.equals("jpg")) || (suffix.equals("jpeg"))) {
+            mimetype = "image/jpeg";
+        }else if (suffix.equals("gif")) {
+            mimetype = "image/gif";
+        }else if (suffix.equals("png")) {
+            mimetype = "image/png";
+        }
+        
+        if (mimetype != "") {
+            ContentResolver contentresolver = this.getContentResolver();
+            ContentValues contentvalues = new ContentValues();
+            contentvalues.put(MediaStore.Images.Media.MIME_TYPE, mimetype);
+            contentvalues.put(MediaStore.Images.Media.DATA, filepath);
+            contentvalues.put(MediaStore.Images.Media.TITLE, filename);
+            contentvalues.put(MediaStore.Images.Media.DISPLAY_NAME, filename);
+            Uri uri = contentresolver.insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, contentvalues);
+            dp(uri.getPath());
+        }else{
+        	dp("Fail: ギャラリーへの登録");
+        }
+    }
+
+    public void unregisterFileToGallery(String name)
+    {
+    	if( name.length()==0 ) return;
+    	
+    	String path = Utility.getRootPath() + "/" + name + ".png";
+		ContentResolver resolver = this.getContentResolver();
+		int affect = resolver.delete(
+				MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
+				MediaStore.Images.Media.DATA + " = '" + path + "'", null);
+		if(affect == 0) dp("Fail: ギャラリーへの抹消");
+    }
+    
+    /*
+     * URI から path へ変換する
+     */
+    private String getUriToPath(Uri uri)
+    {
+		Cursor c = getContentResolver().query(uri, null, null, null, null);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *c*. Currently, there are execution paths that do not contain closure statements. Either a) close *c* in a try-finally block or b) close the resource by declaring *c* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).